### ISSUE TITLE
Access denied issue with workspace directory

### ISSUE THREAD

**Issue by rgov** [jenkinsci/docker-plugin#912]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```text
Jenkins: 2.346.3
OS: Linux - 5.15.0-43-generic
---
ace-editor:1.1
antisamy-markup-formatter:2.7
apache-httpcomponents-client-4-api:4.5.13-138.v4e7d9a_7b_a_e61
authentication-tokens:1.4
azure-ad:267.v5b_dfb_514d9fd
azure-sdk:118.v43f74dd9ca_dc
bootstrap5-api:5.2.0-3
bouncycastle-api:2.26
branch-api:2.1046.v0ca_37783ecc5
build-timeout:1.23
caffeine-api:2.9.3-65.v6a_47d0f4d1fe
checks-api:1.7.5
cloudbees-folder:6.740.ve4f4ffa_dea_54
command-launcher:84.v4a_97f2027398
credentials:1139.veb_9579fca_33b_
credentials-binding:523.vd859a_4b_122e6
display-url-api:2.3.6
docker-commons:1.21
docker-java-api:3.2.13-37.vf3411c9828b9
docker-plugin:1.2.9
durable-task:500.v8927d9fd99d8
echarts-api:5.3.3-1
email-ext:2.91
font-awesome-api:6.1.2-1
generic-webhook-trigger:1.84
git:4.11.5
git-client:3.11.2
github:1.34.5
github-api:1.303-400.v35c2d8258028
github-branch-source:1687.v7618247e672d
handlebars:3.0.8
jackson2-api:2.13.3-285.vc03c0256d517
jakarta-activation-api:2.0.1-1
jakarta-mail-api:2.0.1-1
javax-activation-api:1.2.0-4
javax-mail-api:1.6.2-7
jaxb:2.3.6-1
jdk-tool:55.v1b_32b_6ca_f9ca
jjwt-api:0.11.5-77.v646c772fddb_0
jquery3-api:3.6.0-4
jsch:0.1.55.61.va_e9ee26616e7
junit:1119.1121.vc43d0fc45561
mailer:435.v79ef3972b_5c7
matrix-auth:3.1.5
matrix-project:772.v494f19991984
mina-sshd-api-common:2.8.0-36.v8e25ce90d4b_1
mina-sshd-api-core:2.8.0-36.v8e25ce90d4b_1
momentjs:1.1.1
okhttp-api:4.9.3-108.v0feda04578cf
pam-auth:1.10
pipeline-build-step:2.18
pipeline-github-lib:38.v445716ea_edda_
pipeline-graph-analysis:195.v5812d95a_a_2f9
pipeline-groovy-lib:612.v84da_9c54906d
pipeline-input-step:449.v77f0e8b_845c4
pipeline-milestone-step:101.vd572fef9d926
pipeline-model-api:2.2114.v2654ca_721309
pipeline-model-definition:2.2114.v2654ca_721309
pipeline-model-extensions:2.2114.v2654ca_721309
pipeline-rest-api:2.24
pipeline-stage-step:293.v200037eefcd5
pipeline-stage-tags-metadata:2.2114.v2654ca_721309
pipeline-stage-view:2.24
plain-credentials:139.ved2b_9cf7587b
plugin-util-api:2.17.0
popper2-api:2.11.6-1
resource-disposer:0.19
scm-api:621.vda_a_b_055e58f7
script-security:1175.v4b_d517d6db_f0
snakeyaml-api:1.30.2-76.vc104f7ce9870
ssh-credentials:295.vced876c18eb_4
ssh-slaves:1.834.v622da_57f702c
sshd:3.242.va_db_9da_b_26a_c3
structs:324.va_f5d6774f3a_d
timestamper:1.18
token-macro:308.v4f2b_ed62b_b_16
trilead-api:1.67.vc3938a_35172f
variant:59.vf075fe829ccb
workflow-aggregator:590.v6a_d052e5a_a_b_5
workflow-api:1192.v2d0deb_19d212
workflow-basic-steps:991.v43d80fea_ff66
workflow-cps:2759.v87459c4eea_ca_
workflow-durable-task-step:1199.v02b_9244f8064
workflow-job:1207.ve6191ff089f8
workflow-multibranch:716.vc692a_e52371b_
workflow-scm-step:400.v6b_89a_1317c9a_
workflow-step-api:639.v6eca_cd8c04a_a_
workflow-support:838.va_3a_087b_4055b
ws-cleanup:0.42
```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

Host is Ubuntu 22.04, master is running in a (modified) `jenkins/jenkins:lts` container, agent is running in a `jenkins/inbound-agent` container.

### Reproduction steps

1. Build a Jenkins master container image based on the `jenkins/jenkins:lts` image, adding the `jenkins` user to the `docker` group, whose GID must match the host's `docker` group.

2. Run the Jenkins master container, mounting the host's Docker Engine socket at `/var/run/docker.sock` into the container. Because the `jenkins` user in the container is in the `docker` group, it is permitted to connect to the socket.

3. Configure the Docker plugin to use the Unix socket that has been mounted into the container. Confirm that this works.

4. Configure the agent template to use the `jenkins/inbound-agent` image with the default `jenkins` user.

5. Build a project against a Docker cloud node using this agent template.

### Expected Results

It gets as far as running the build commands.

### Actual Results

The build agent cannot access the `/workspace` directory as the unprivileged `jenkins` user, throwing a `java.nio.file.AccessDeniedException: /workspace`.

<details>
<summary>Complete build log</summary>

```
Running as SYSTEM
Building remotely on [docker-000r4gfe9hsk0 on Docker](https://buildbot/jenkins/computer/docker-000r4gfe9hsk0) (docker) in workspace /workspace/Sentry
Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to JNLP4-connect connection from 2be1d81dc15c.jenkins_default/172.24.0.3:50062
		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1784)
		at hudson.remoting.UserRequest$ExceptionResponse.retrieve(UserRequest.java:356)
		at hudson.remoting.Channel.call(Channel.java:1000)
		at hudson.FilePath.act(FilePath.java:1194)
		at hudson.FilePath.act(FilePath.java:1183)
		at hudson.FilePath.mkdirs(FilePath.java:1374)
		at hudson.model.AbstractProject.checkout(AbstractProject.java:1211)
		at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:647)
		at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:85)
		at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:519)
		at hudson.model.Run.execute(Run.java:1897)
		at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:44)
		at hudson.model.ResourceController.execute(ResourceController.java:101)
		at hudson.model.Executor.run(Executor.java:442)
java.nio.file.AccessDeniedException: /workspace
	at java.base/sun.nio.fs.UnixException.translateToIOException(Unknown Source)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source)
	at java.base/sun.nio.fs.UnixFileSystemProvider.createDirectory(Unknown Source)
	at java.base/java.nio.file.Files.createDirectory(Unknown Source)
	at java.base/java.nio.file.Files.createAndCheckIsDirectory(Unknown Source)
	at java.base/java.nio.file.Files.createDirectories(Unknown Source)
	at hudson.FilePath.mkdirs(FilePath.java:3624)
	at hudson.FilePath.access$1100(FilePath.java:212)
	at hudson.FilePath$Mkdirs.invoke(FilePath.java:1384)
	at hudson.FilePath$Mkdirs.invoke(FilePath.java:1379)
	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:3502)
	at hudson.remoting.UserRequest.perform(UserRequest.java:211)
	at hudson.remoting.UserRequest.perform(UserRequest.java:54)
	at hudson.remoting.Request$2.run(Request.java:376)
	at hudson.remoting.InterceptingExecutorService.lambda$wrap$0(InterceptingExecutorService.java:78)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at hudson.remoting.Engine$1.lambda$newThread$0(Engine.java:122)
	at java.base/java.lang.Thread.run(Unknown Source)
Finished: FAILURE
```
</details>

By running the agent as the `root` user, I can see that `/workspace` and the files under it are owned by `root`, even though we have been using the unprivileged `jenkins` user throughout. (The `jenkins` user in the Jenkins master image and in the agent image both use UID 1000.)

### Anything else?

I confirmed that my Jenkins home directory and everything under it is owned by `jenkins`. I purged all old Docker volumes in case something was being reused.

Looking at the `docker inspect` output associated with the agent node for my failed build, I do not see `workspace` in the list of mounts for the container. Therefore I don't know how it is getting mounted into the container.

<details>
<summary>`docker inspect` output</summary>

```
{
  "Args" : [ "-url", "http://jenkins:8080/jenkins/", "85d59ac18d33260762c2641aac9e5f7a5056529c94b8cd5f1679f8d07a944c39", "docker-000r4gfe9hsk0" ],
  "Config" : {
    "healthcheck" : null,
    "AttachStderr" : false,
    "AttachStdin" : false,
    "AttachStdout" : false,
    "Cmd" : [ "-url", "http://jenkins:8080/jenkins/", "85d59ac18d33260762c2641aac9e5f7a5056529c94b8cd5f1679f8d07a944c39", "docker-000r4gfe9hsk0" ],
    "Domainname" : "",
    "Entrypoint" : [ "/usr/local/bin/jenkins-agent" ],
    "Env" : [ "PATH=/opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", "LANG=C.UTF-8", "JAVA_HOME=/opt/java/openjdk", "AGENT_WORKDIR=/home/jenkins/agent" ],
    "ExposedPorts" : null,
    "Hostname" : "2be1d81dc15c",
    "Image" : "jenkins/inbound-agent",
    "Labels" : {
      "Description" : "This is a base image, which allows connecting Jenkins agents via JNLP protocols",
      "Vendor" : "Jenkins project",
      "Version" : "4.10-6-jdk11",
      "com.nirima.jenkins.plugins.docker.JenkinsContainerImage" : "jenkins/inbound-agent",
      "com.nirima.jenkins.plugins.docker.JenkinsId" : "ac85cd7d255bce68e5ff5edcecb7ebde",
      "com.nirima.jenkins.plugins.docker.JenkinsNodeName" : "docker-000r4gfe9hsk0",
      "com.nirima.jenkins.plugins.docker.JenkinsRemoveVolumes" : "false",
      "com.nirima.jenkins.plugins.docker.JenkinsServerUrl" : "https://buildbot/jenkins/",
      "com.nirima.jenkins.plugins.docker.JenkinsTemplateName" : "docker",
      "org.opencontainers.image.description" : "This is a base image, which provides the Jenkins agent executable (agent.jar)",
      "org.opencontainers.image.licenses" : "MIT",
      "org.opencontainers.image.source" : "https://github.com/jenkinsci/docker-agent",
      "org.opencontainers.image.title" : "Official Jenkins Agent Base Docker image",
      "org.opencontainers.image.url" : "https://www.jenkins.io/",
      "org.opencontainers.image.vendor" : "Jenkins project",
      "org.opencontainers.image.version" : "4.10"
    },
    "MacAddress" : null,
    "NetworkDisabled" : null,
    "OnBuild" : null,
    "OpenStdin" : false,
    "PortSpecs" : null,
    "StdinOnce" : false,
    "Tty" : false,
    "User" : "jenkins",
    "Volumes" : {
      "/home/jenkins/.jenkins" : { },
      "/home/jenkins/agent" : { }
    },
    "WorkingDir" : "/home/jenkins",
    "Healthcheck" : null
  },
  "Created" : "2022-08-30T18:12:01.473309125Z",
  "Driver" : "overlay2",
  "ExecDriver" : null,
  "HostConfig" : {
    "Binds" : null,
    "BlkioWeight" : 0,
    "BlkioWeightDevice" : null,
    "BlkioDeviceReadBps" : null,
    "BlkioDeviceWriteBps" : null,
    "BlkioDeviceReadIOps" : null,
    "BlkioDeviceWriteIOps" : null,
    "MemorySwappiness" : null,
    "NanoCpus" : 0,
    "CapAdd" : null,
    "CapDrop" : null,
    "ContainerIDFile" : "",
    "CpuPeriod" : 0,
    "CpuRealtimePeriod" : 0,
    "CpuRealtimeRuntime" : 0,
    "CpuShares" : 0,
    "CpuQuota" : 0,
    "CpusetCpus" : "",
    "CpusetMems" : "",
    "Devices" : null,
    "DeviceCgroupRules" : null,
    "DeviceRequests" : null,
    "DiskQuota" : null,
    "Dns" : null,
    "DnsOptions" : null,
    "DnsSearch" : null,
    "ExtraHosts" : null,
    "GroupAdd" : null,
    "IpcMode" : "private",
    "Cgroup" : "",
    "Links" : null,
    "LogConfig" : {
      "Type" : "json-file",
      "Config" : { }
    },
    "LxcConf" : null,
    "Memory" : 0,
    "MemorySwap" : 0,
    "MemoryReservation" : 0,
    "KernelMemory" : 0,
    "NetworkMode" : "jenkins_default",
    "OomKillDisable" : null,
    "Init" : null,
    "AutoRemove" : false,
    "OomScoreAdj" : 0,
    "PortBindings" : { },
    "Privileged" : false,
    "PublishAllPorts" : false,
    "ReadonlyRootfs" : false,
    "RestartPolicy" : {
      "MaximumRetryCount" : 0,
      "Name" : ""
    },
    "Ulimits" : null,
    "CpuCount" : 0,
    "CpuPercent" : 0,
    "IOMaximumIOps" : 0,
    "IOMaximumBandwidth" : 0,
    "VolumesFrom" : null,
    "Mounts" : null,
    "PidMode" : "",
    "Isolation" : null,
    "SecurityOpt" : null,
    "StorageOpt" : null,
    "CgroupParent" : "",
    "VolumeDriver" : "",
    "ShmSize" : 67108864,
    "PidsLimit" : null,
    "Runtime" : "runc",
    "Tmpfs" : null,
    "UTSMode" : "",
    "UsernsMode" : "",
    "Sysctls" : null,
    "ConsoleSize" : [ 0, 0 ],
    "CgroupnsMode" : "private"
  },
  "HostnamePath" : "/var/lib/docker/containers/2be1d81dc15c1f24d29a758d8f87596fc9f0229c47de46d991fbb0f3d076f2eb/hostname",
  "HostsPath" : "/var/lib/docker/containers/2be1d81dc15c1f24d29a758d8f87596fc9f0229c47de46d991fbb0f3d076f2eb/hosts",
  "LogPath" : "/var/lib/docker/containers/2be1d81dc15c1f24d29a758d8f87596fc9f0229c47de46d991fbb0f3d076f2eb/2be1d81dc15c1f24d29a758d8f87596fc9f0229c47de46d991fbb0f3d076f2eb-json.log",
  "Id" : "2be1d81dc15c1f24d29a758d8f87596fc9f0229c47de46d991fbb0f3d076f2eb",
  "SizeRootFs" : null,
  "Image" : "sha256:e790c88afef25551ae6b4210d742decc1c7d34712f1b98db7959fef5b4f404fd",
  "MountLabel" : "",
  "Name" : "/zen_pare",
  "RestartCount" : 0,
  "NetworkSettings" : {
    "Bridge" : "",
    "SandboxID" : "49f32399cbdcfb420fcd55d905b4bd33ae71a40084bda9e2304dec2821a4e32e",
    "HairpinMode" : false,
    "LinkLocalIPv6Address" : "",
    "LinkLocalIPv6PrefixLen" : 0,
    "Ports" : { },
    "SandboxKey" : "/var/run/docker/netns/49f32399cbdc",
    "SecondaryIPAddresses" : null,
    "SecondaryIPv6Addresses" : null,
    "EndpointID" : "",
    "Gateway" : "",
    "PortMapping" : null,
    "GlobalIPv6Address" : "",
    "GlobalIPv6PrefixLen" : 0,
    "IPAddress" : "",
    "IPPrefixLen" : 0,
    "IPv6Gateway" : "",
    "MacAddress" : "",
    "Networks" : {
      "jenkins_default" : {
        "IPAMConfig" : null,
        "Links" : null,
        "Aliases" : [ "2be1d81dc15c" ],
        "NetworkID" : "54f22865b1f6fbad1d3ccfedaa6a66a39ca4d5f7fa6c4a3dbee1121ef58611c4",
        "EndpointID" : "43521839cec283848d7f596261d9412c004ecdbad7ec50dca470828fcd674cbe",
        "Gateway" : "172.24.0.1",
        "IPAddress" : "172.24.0.3",
        "IPPrefixLen" : 16,
        "IPv6Gateway" : "",
        "GlobalIPv6Address" : "",
        "GlobalIPv6PrefixLen" : 0,
        "MacAddress" : "02:42:ac:18:00:03"
      }
    }
  },
  "Path" : "/usr/local/bin/jenkins-agent",
  "ProcessLabel" : "",
  "ResolvConfPath" : "/var/lib/docker/containers/2be1d81dc15c1f24d29a758d8f87596fc9f0229c47de46d991fbb0f3d076f2eb/resolv.conf",
  "ExecIDs" : null,
  "State" : {
    "oomkilled" : false,
    "pidLong" : 3111285,
    "exitCodeLong" : 0,
    "Status" : "running",
    "Running" : true,
    "Paused" : false,
    "Restarting" : false,
    "OOMKilled" : false,
    "Dead" : false,
    "Pid" : 3111285,
    "ExitCode" : 0,
    "Error" : "",
    "StartedAt" : "2022-08-30T18:12:01.859902899Z",
    "FinishedAt" : "0001-01-01T00:00:00Z",
    "Health" : null
  },
  "Volumes" : null,
  "VolumesRW" : null,
  "Node" : null,
  "Mounts" : [ {
    "Name" : "4a2b3e836be7c9ade3f91d269511367877a227d89b8a662c3f4399946b1c2eb8",
    "Source" : "/var/lib/docker/volumes/4a2b3e836be7c9ade3f91d269511367877a227d89b8a662c3f4399946b1c2eb8/_data",
    "Destination" : "/home/jenkins/.jenkins",
    "Driver" : "local",
    "Mode" : "",
    "RW" : true
  }, {
    "Name" : "7471f1be80f44c6c7ffe8272bea0601b645bf949e6b6c19e1d0fb2de75e83b79",
    "Source" : "/var/lib/docker/volumes/7471f1be80f44c6c7ffe8272bea0601b645bf949e6b6c19e1d0fb2de75e83b79/_data",
    "Destination" : "/home/jenkins/agent",
    "Driver" : "local",
    "Mode" : "",
    "RW" : true
  } ],
  "GraphDriver" : {
    "Name" : "overlay2",
    "Data" : {
      "RootDir" : null,
      "DeviceId" : null,
      "DeviceName" : null,
      "DeviceSize" : null,
      "dir" : null
    }
  },
  "Platform" : "linux",
  "Path" : "/usr/local/bin/jenkins-agent",
  "Args" : [ "-url", "http://jenkins:8080/jenkins/", "85d59ac18d33260762c2641aac9e5f7a5056529c94b8cd5f1679f8d07a944c39", "docker-000r4gfe9hsk0" ],
  "ProcessLabel" : "",
  "Mounts" : [ {
    "Type" : "volume",
    "Name" : "4a2b3e836be7c9ade3f91d269511367877a227d89b8a662c3f4399946b1c2eb8",
    "Source" : "/var/lib/docker/volumes/4a2b3e836be7c9ade3f91d269511367877a227d89b8a662c3f4399946b1c2eb8/_data",
    "Destination" : "/home/jenkins/.jenkins",
    "Driver" : "local",
    "Mode" : "",
    "RW" : true,
    "Propagation" : ""
  }, {
    "Type" : "volume",
    "Name" : "7471f1be80f44c6c7ffe8272bea0601b645bf949e6b6c19e1d0fb2de75e83b79",
    "Source" : "/var/lib/docker/volumes/7471f1be80f44c6c7ffe8272bea0601b645bf949e6b6c19e1d0fb2de75e83b79/_data",
    "Destination" : "/home/jenkins/agent",
    "Driver" : "local",
    "Mode" : "",
    "RW" : true,
    "Propagation" : ""
  } ],
  "Platform" : "linux",
  "Config" : {
    "Hostname" : "2be1d81dc15c",
    "Domainname" : "",
    "User" : "jenkins",
    "AttachStdin" : false,
    "AttachStdout" : false,
    "AttachStderr" : false,
    "Tty" : false,
    "OpenStdin" : false,
    "StdinOnce" : false,
    "Env" : [ "PATH=/opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", "LANG=C.UTF-8", "JAVA_HOME=/opt/java/openjdk", "AGENT_WORKDIR=/home/jenkins/agent" ],
    "Cmd" : [ "-url", "http://jenkins:8080/jenkins/", "85d59ac18d33260762c2641aac9e5f7a5056529c94b8cd5f1679f8d07a944c39", "docker-000r4gfe9hsk0" ],
    "Image" : "jenkins/inbound-agent",
    "Volumes" : {
      "/home/jenkins/.jenkins" : { },
      "/home/jenkins/agent" : { }
    },
    "WorkingDir" : "/home/jenkins",
    "Entrypoint" : [ "/usr/local/bin/jenkins-agent" ],
    "OnBuild" : null,
    "Labels" : {
      "Description" : "This is a base image, which allows connecting Jenkins agents via JNLP protocols",
      "Vendor" : "Jenkins project",
      "Version" : "4.10-6-jdk11",
      "com.nirima.jenkins.plugins.docker.JenkinsContainerImage" : "jenkins/inbound-agent",
      "com.nirima.jenkins.plugins.docker.JenkinsId" : "ac85cd7d255bce68e5ff5edcecb7ebde",
      "com.nirima.jenkins.plugins.docker.JenkinsNodeName" : "docker-000r4gfe9hsk0",
      "com.nirima.jenkins.plugins.docker.JenkinsRemoveVolumes" : "false",
      "com.nirima.jenkins.plugins.docker.JenkinsServerUrl" : "https://buildbot/jenkins/",
      "com.nirima.jenkins.plugins.docker.JenkinsTemplateName" : "docker",
      "org.opencontainers.image.description" : "This is a base image, which provides the Jenkins agent executable (agent.jar)",
      "org.opencontainers.image.licenses" : "MIT",
      "org.opencontainers.image.source" : "https://github.com/jenkinsci/docker-agent",
      "org.opencontainers.image.title" : "Official Jenkins Agent Base Docker image",
      "org.opencontainers.image.url" : "https://www.jenkins.io/",
      "org.opencontainers.image.vendor" : "Jenkins project",
      "org.opencontainers.image.version" : "4.10"
    }
  },
  "Driver" : "overlay2",
  "AppArmorProfile" : "docker-default",
  "HostsPath" : "/var/lib/docker/containers/2be1d81dc15c1f24d29a758d8f87596fc9f0229c47de46d991fbb0f3d076f2eb/hosts",
  "HostnamePath" : "/var/lib/docker/containers/2be1d81dc15c1f24d29a758d8f87596fc9f0229c47de46d991fbb0f3d076f2eb/hostname",
  "GraphDriver" : {
    "Data" : {
      "LowerDir" : "/var/lib/docker/overlay2/a17950c47f018c22a916b5bd0344237cafb32f9f6de6816f04b2004d71f9b4a9-init/diff:/var/lib/docker/overlay2/ccc863815495a8835d223d9038c0711e82d80743a631c31a557c452f02739d34/diff:/var/lib/docker/overlay2/603e76c60be4bce3498b39157142b9781992146598b205caa0ef7d648b1da624/diff:/var/lib/docker/overlay2/d9cc3f236bb4a42bfd6c93102bbc08d4aecec099a14df0467d3533355b1bb802/diff:/var/lib/docker/overlay2/026bbc64d9e5853a9dc9f07f735866c6c0b9eeed467b74426ddaea239c86d8c3/diff:/var/lib/docker/overlay2/2df555241aeb17e48a1ebb967273f0ec417ffb2c0ea6d3f29de56533b0d55095/diff:/var/lib/docker/overlay2/c4e9498798e1c6df07b10b0c92f8d0469e4e9589ef25d67f8c61f34b43da0cb0/diff:/var/lib/docker/overlay2/28fe696938ace2a7f32c7d5cf23729fa8854b0d0d0ed22dbc7a54a20677cfe41/diff:/var/lib/docker/overlay2/f3516b4aac1cfa3276872f4972aa1b03644ad9cbea87a6ea5dc85299fffef5e8/diff:/var/lib/docker/overlay2/7cc1e94530fdd885402cf9af4057e879f8fde7a09a9672671a731b4feaa2dedb/diff",
      "MergedDir" : "/var/lib/docker/overlay2/a17950c47f018c22a916b5bd0344237cafb32f9f6de6816f04b2004d71f9b4a9/merged",
      "UpperDir" : "/var/lib/docker/overlay2/a17950c47f018c22a916b5bd0344237cafb32f9f6de6816f04b2004d71f9b4a9/diff",
      "WorkDir" : "/var/lib/docker/overlay2/a17950c47f018c22a916b5bd0344237cafb32f9f6de6816f04b2004d71f9b4a9/work"
    },
    "Name" : "overlay2"
  },
  "Image" : "sha256:e790c88afef25551ae6b4210d742decc1c7d34712f1b98db7959fef5b4f404fd",
  "Created" : "2022-08-30T18:12:01.473309125Z",
  "Name" : "/zen_pare",
  "RestartCount" : 0,
  "NetworkSettings" : {
    "Bridge" : "",
    "SandboxID" : "49f32399cbdcfb420fcd55d905b4bd33ae71a40084bda9e2304dec2821a4e32e",
    "HairpinMode" : false,
    "LinkLocalIPv6Address" : "",
    "LinkLocalIPv6PrefixLen" : 0,
    "Ports" : { },
    "SandboxKey" : "/var/run/docker/netns/49f32399cbdc",
    "SecondaryIPAddresses" : null,
    "SecondaryIPv6Addresses" : null,
    "EndpointID" : "",
    "Gateway" : "",
    "GlobalIPv6Address" : "",
    "GlobalIPv6PrefixLen" : 0,
    "IPAddress" : "",
    "IPPrefixLen" : 0,
    "IPv6Gateway" : "",
    "MacAddress" : "",
    "Networks" : {
      "jenkins_default" : {
        "IPAMConfig" : null,
        "Links" : null,
        "Aliases" : [ "2be1d81dc15c" ],
        "NetworkID" : "54f22865b1f6fbad1d3ccfedaa6a66a39ca4d5f7fa6c4a3dbee1121ef58611c4",
        "EndpointID" : "43521839cec283848d7f596261d9412c004ecdbad7ec50dca470828fcd674cbe",
        "Gateway" : "172.24.0.1",
        "IPAddress" : "172.24.0.3",
        "IPPrefixLen" : 16,
        "IPv6Gateway" : "",
        "GlobalIPv6Address" : "",
        "GlobalIPv6PrefixLen" : 0,
        "MacAddress" : "02:42:ac:18:00:03",
        "DriverOpts" : null
      }
    }
  },
  "MountLabel" : "",
  "State" : {
    "Status" : "running",
    "Running" : true,
    "Paused" : false,
    "Restarting" : false,
    "OOMKilled" : false,
    "Dead" : false,
    "Pid" : 3111285,
    "ExitCode" : 0,
    "Error" : "",
    "StartedAt" : "2022-08-30T18:12:01.859902899Z",
    "FinishedAt" : "0001-01-01T00:00:00Z"
  },
  "ResolvConfPath" : "/var/lib/docker/containers/2be1d81dc15c1f24d29a758d8f87596fc9f0229c47de46d991fbb0f3d076f2eb/resolv.conf",
  "ExecIDs" : null,
  "HostConfig" : {
    "Binds" : null,
    "ContainerIDFile" : "",
    "LogConfig" : {
      "Type" : "json-file",
      "Config" : { }
    },
    "NetworkMode" : "jenkins_default",
    "PortBindings" : { },
    "RestartPolicy" : {
      "Name" : "",
      "MaximumRetryCount" : 0
    },
    "AutoRemove" : false,
    "VolumeDriver" : "",
    "VolumesFrom" : null,
    "CapAdd" : null,
    "CapDrop" : null,
    "CgroupnsMode" : "private",
    "Dns" : null,
    "DnsOptions" : null,
    "DnsSearch" : null,
    "ExtraHosts" : null,
    "GroupAdd" : null,
    "IpcMode" : "private",
    "Cgroup" : "",
    "Links" : null,
    "OomScoreAdj" : 0,
    "PidMode" : "",
    "Privileged" : false,
    "PublishAllPorts" : false,
    "ReadonlyRootfs" : false,
    "SecurityOpt" : null,
    "UTSMode" : "",
    "UsernsMode" : "",
    "ShmSize" : 67108864,
    "Runtime" : "runc",
    "ConsoleSize" : [ 0, 0 ],
    "Isolation" : "",
    "CpuShares" : 0,
    "Memory" : 0,
    "NanoCpus" : 0,
    "CgroupParent" : "",
    "BlkioWeight" : 0,
    "BlkioWeightDevice" : null,
    "BlkioDeviceReadBps" : null,
    "BlkioDeviceWriteBps" : null,
    "BlkioDeviceReadIOps" : null,
    "BlkioDeviceWriteIOps" : null,
    "CpuPeriod" : 0,
    "CpuQuota" : 0,
    "CpuRealtimePeriod" : 0,
    "CpuRealtimeRuntime" : 0,
    "CpusetCpus" : "",
    "CpusetMems" : "",
    "Devices" : null,
    "DeviceCgroupRules" : null,
    "DeviceRequests" : null,
    "KernelMemory" : 0,
    "KernelMemoryTCP" : 0,
    "MemoryReservation" : 0,
    "MemorySwap" : 0,
    "MemorySwappiness" : null,
    "OomKillDisable" : null,
    "PidsLimit" : null,
    "Ulimits" : null,
    "CpuCount" : 0,
    "CpuPercent" : 0,
    "IOMaximumIOps" : 0,
    "IOMaximumBandwidth" : 0,
    "MaskedPaths" : [ "/proc/asound", "/proc/acpi", "/proc/kcore", "/proc/keys", "/proc/latency_stats", "/proc/timer_list", "/proc/timer_stats", "/proc/sched_debug", "/proc/scsi", "/sys/firmware" ],
    "ReadonlyPaths" : [ "/proc/bus", "/proc/fs", "/proc/irq", "/proc/sys", "/proc/sysrq-trigger" ]
  },
  "Id" : "2be1d81dc15c1f24d29a758d8f87596fc9f0229c47de46d991fbb0f3d076f2eb",
  "LogPath" : "/var/lib/docker/containers/2be1d81dc15c1f24d29a758d8f87596fc9f0229c47de46d991fbb0f3d076f2eb/2be1d81dc15c1f24d29a758d8f87596fc9f0229c47de46d991fbb0f3d076f2eb-json.log"
}
```
</details>

**Comment by rgov** [comment]
I think it was just trying to `mkdir /workspace` and this was failing as a non-root user because it can't create a top-level directories. The solution I found was to configure the "Remote File System Root" setting to `/home/jenkins/agent` which already exists in the container image and is writable by the `jenkins` user.

However the name and description of this setting are not clear to me, especially with separate settings for the "remoting workdir" and then the container's own working directory setting. Maybe it should be something like "Container workspace base directory"?

**Comment by xabolcs** [comment]
Another workaround is to use `jenkins/inbound-agent:4.9-1`. That old version works nicely with empty "Remote File System Root" value.

**Comment by xabolcs** [comment]
#867 points to the real cause of this.

**Comment by basil** [comment]
Fixed in #942.
